src/frontend/src/components/modal.ts

```diff
@@ -0,0 +1,59 @@
+import { html, TemplateResult, render } from "lit-html";
+
+export const createModal = ({
+  slot,
+  id,
+  confirmText = "Ok",
+}: {
+  slot: TemplateResult;
+  id?: string;
+  confirmText?: string;
+}): HTMLDialogElement => {
+  const modalHtml = html`
+    <form method="dialog" class="c-modal__content c-card c-card--modal">
+      <div class="c-modal__inner">
+        <span class="c-modal__close" data-closemodal>&times;</span>
+        ${slot}
+        <div class="c-modal__footer">
+          <button type="submit" class="c-button c-button--primary">
+            ${confirmText}
+          </button>
+        </div>
+      </div>
+    </form>
+  `;
+
+  const $modal = document.createElement("dialog");
+  $modal.id = id != null ? id : "modal";
+  $modal.classList.add("c-modal");
+  $modal.setAttribute("aria-modal", "true");
```

Comment:
Why mix `render` and `createElement`? Why not return a `TemplateResult` instead?

Reply:
Because it needs to return a native DOM element that I can append to the body. How wo that work with TempleteResult?